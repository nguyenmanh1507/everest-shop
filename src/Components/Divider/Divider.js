// @flow

import React, { PureComponent } from 'react'
import cx from 'classnames'

type DividerTypeSuit = 'default' | 'dot' | 'dark'

type Props = {
  /** Divider Type - Accept value: 'default', 'dot', 'dark' */
  type: DividerTypeSuit,
  /** Extra class */
  exClass?: string
}

class Divider extends PureComponent<Props> {
  static defaultProps = {
    type: 'default'
  }

  render() {
    const { type, exClass } = this.props

    return (
      <div
        className={cx({
          divider: type === 'default',
          'divider-dot': type === 'dot',
          'divider-dark': type === 'dark',
          [exClass || '']: exClass && exClass.length
        })}
      />
    )
  }
}

export default Divider
