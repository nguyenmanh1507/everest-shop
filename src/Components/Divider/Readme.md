Divider example:

### Default

```js
<Divider />
```

### Dot

```js
<Divider type="dot" />
```

### Dark

```js
<Divider type="dark" />
```
