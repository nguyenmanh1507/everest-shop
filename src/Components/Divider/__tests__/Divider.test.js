// @flow

import React from 'react'
import renderer from 'react-test-renderer'
import { render } from 'enzyme'

import Divider from '../Divider'

test('Divider render properly', () => {
  const component = renderer.create(<Divider />)

  let tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})

test('Divider render type dot', () => {
  const component = renderer.create(<Divider type="dot" />)

  let tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})

test('Divider render type dark', () => {
  const component = renderer.create(<Divider type="dark" />)

  let tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})

test('Divider with extra class', () => {
  const component = renderer.create(<Divider type="dot" exClass="my-md" />)

  let tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})
