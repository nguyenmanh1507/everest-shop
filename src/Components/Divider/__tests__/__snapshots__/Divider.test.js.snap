// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Divider render properly 1`] = `
<div
  className="divider"
/>
`;

exports[`Divider render type dark 1`] = `
<div
  className="divider-dark"
/>
`;

exports[`Divider render type dot 1`] = `
<div
  className="divider-dot"
/>
`;

exports[`Divider with extra class 1`] = `
<div
  className="divider-dot my-md"
/>
`;
